src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -114,13 +119,15 @@
   }
 
   // parse the given file into a List of CastVoteRecords for tabulation
-  void parseCvrFile(List<CastVoteRecord> castVoteRecords) {
+  void parseCvrFile(List<CastVoteRecord> castVoteRecords) throws UnrecognizedCandidatesException {
     // cvrIndex and fileName are used to generate IDs for cvrs
     int cvrIndex = 0;
     String fileName = new File(filePath).getName();
 
     try {
       HashMap json = JsonParser.readFromFile(filePath, HashMap.class);
+      Map<String, String> candidates = getCandidates();
```

Comment:
Nit: you can just inline this.

Reply:
That would be very inefficient -- it would mean that we were reloading the whole CVR to get the candidates again for each ballot that we process.

Reply:
Ah, right you are, I missed the loop due to GitHub review line hiding.